### PR TITLE
Remove the legacy 'maven' plugin

### PR DESCRIPTION
**Pull request by melix**
### Context

This pull request removes the legacy "maven" plugin in favor of the "maven-publish" plugin, aka Grapocalypse part 2.

There are examples which are now harder to do, we should consider making those easier (see #15606 ).

### CODE REVIEW

**Comment by melix** [comment]
@bot-gradle test RFM

**Comment by melix** [comment]
@bot-gradle test RFM

**Comment by melix** [comment]
@bot-gradle test RFM

**Comment by melix** [comment]
@bot-gradle test RFM

**Comment by melix** [comment]
@bot-gradle test this

**Comment by melix** [comment]
@bot-gradle test this

**Comment by melix** [comment]
@bot-gradle test this

**Comment by blindpirate** [comment]
@bot-gradle test this

**Comment by melix** [comment]
@bot-gradle test this

**Comment by melix** [comment]
@bot-gradle test this

**Comment by melix** [comment]
@bot-gradle test this

File: subprojects/plugins/src/main/java/org/gradle/api/plugins/MavenPlugin.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    private final DocumentationRegistry documentationRegistry;
+
+    @Inject
+    public MavenPlugin(DocumentationRegistry documentationRegistry) {
+        this.documentationRegistry = documentationRegistry;
+    }
+
+    @Override
+    public void apply(Project target) {
+        throw new UnsupportedOperationException(
```
Comment (ljacomet) [inline subprojects/plugins/src/main/java/org/gradle/api/plugins/MavenPlugin.java:39]: Should you not use `InvalidUserCodeException` here?

**Review by ljacomet — APPROVED** [review]
LGTM, thanks for doing the cleanup.

One minor comment on an exception type, see below.

**Pull request merged**